tests/functional/olp-cpp-sdk-dataservice-read/CatalogClientTest.cpp

```diff
@@ -0,0 +1,598 @@
+            dataResponse.GetError().GetErrorCode());
+}
+
+TEST_P(CatalogClientTest, GetDataWithInlineField) {
+  olp::client::HRN hrn(GetTestCatalog());
+
+  auto catalogClient =
+      std::make_unique<olp::dataservice::read::CatalogClient>(hrn, settings_);
+
+  auto request = olp::dataservice::read::DataRequest();
+  request.WithLayerId("testlayer").WithPartitionId("3");
+  auto dataResponse =
+      GetExecutionTime<olp::dataservice::read::DataResponse>([&] {
+        auto future = catalogClient->GetData(request);
+        return future.GetFuture().get();
+      });
+
+  ASSERT_TRUE(dataResponse.IsSuccessful())
+      << PrintError(dataResponse.GetError());
+  ASSERT_LT(0, dataResponse.GetResult()->size());
+  std::string dataStr(dataResponse.GetResult()->begin(),
+                      dataResponse.GetResult()->end());
+  ASSERT_EQ(0u, dataStr.find("data:"));
+}
+
+TEST_P(CatalogClientTest, GetDataWithEmptyField) {
+  olp::client::HRN hrn(GetTestCatalog());
+
+  auto catalogClient =
+      std::make_unique<olp::dataservice::read::CatalogClient>(hrn, settings_);
+
+  auto request = olp::dataservice::read::DataRequest();
+  request.WithLayerId("testlayer").WithPartitionId("1");
+  auto dataResponse =
+      GetExecutionTime<olp::dataservice::read::DataResponse>([&] {
+        auto future = catalogClient->GetData(request);
+        return future.GetFuture().get();
+      });
+
+  ASSERT_TRUE(dataResponse.IsSuccessful())
+      << PrintError(dataResponse.GetError());
+  ASSERT_FALSE(dataResponse.GetResult());
+}
+
+TEST_P(CatalogClientTest, GetDataCompressed) {
+  olp::client::HRN hrn(GetTestCatalog());
+
+  auto catalogClient =
+      std::make_unique<olp::dataservice::read::CatalogClient>(hrn, settings_);
+
+  auto request = olp::dataservice::read::DataRequest();
+  request.WithLayerId("testlayer").WithPartitionId("here_van_wc2018_pool");
+  auto dataResponse =
+      GetExecutionTime<olp::dataservice::read::DataResponse>([&] {
+        auto future = catalogClient->GetData(request);
+        return future.GetFuture().get();
+      });
+
+  ASSERT_TRUE(dataResponse.IsSuccessful())
+      << PrintError(dataResponse.GetError());
+  ASSERT_LT(0u, dataResponse.GetResult()->size());
+
+  auto requestCompressed = olp::dataservice::read::DataRequest();
+  requestCompressed.WithLayerId("testlayer_gzip")
+      .WithPartitionId("here_van_wc2018_pool");
+  auto dataResponseCompressed =
+      GetExecutionTime<olp::dataservice::read::DataResponse>([&] {
+        auto future = catalogClient->GetData(requestCompressed);
+        return future.GetFuture().get();
+      });
+
+  ASSERT_TRUE(dataResponseCompressed.IsSuccessful())
+      << PrintError(dataResponseCompressed.GetError());
+  ASSERT_LT(0u, dataResponseCompressed.GetResult()->size());
+  ASSERT_EQ(dataResponse.GetResult()->size(),
+            dataResponseCompressed.GetResult()->size());
+}
+
+void dumpTileKey(const olp::geo::TileKey& tileKey) {
```

Comment:
Move to anonymous namespace.
```suggestion
void DumpTileKey(const olp::geo::TileKey& tileKey) {
```